src/BigBlueButton.php

```diff
@@ -90,6 +97,51 @@ public function getApiVersion()
         return new ApiVersionResponse($xml);
     }
 
+    /**
+     * Check if connection to api can be established with the baseurl and secret
+     * @return bool connection successful
+     */
+    public function isConnectionWorking(): bool
+    {
+        $this->connectionError = null;
+        try {
+            $response = $this->isMeetingRunning(
+                new IsMeetingRunningParameters("connection_check")
+            );
+
+            // invalid url
+            if (!$response->success() && !$response->failed()) {
+                $this->connectionError = self::CONNECTION_ERROR_BASEURL;
+                return false;
+            }
+
+            // invalid secret
+            if (!$response->success()) {
+                $this->connectionError = self::CONNECTION_ERROR_SECRET;
+                return false;
+            }
+
+           // url and secret are valid
+           if($response->success()) {
+               return true;
```

Comment:
We should reset connectionError in case an application is retrying after a timeout.

Reply:
> I'm still not sure if cs-fixer is satisfied, but the code looks good, except the two things I mentioned.

cs-fixer was used on the latest commit

Reply:
> We should reset connectionError in case an application is retrying after a timeout.

The connectionError is reseted at the beginning of the function.